### DISCUSSION TITLE
Just the compiler

### DISCUSSION THREAD

**Discussion by reggi** [microsoft/devicescript#687] [General]
I was looking at this post: https://www.reddit.com/r/typescript/comments/sphnzx/is_it_possible_to_compile_typescript_directly_to/

Came across this paper and recognized some of the names 😸  https://www.microsoft.com/en-us/research/publication/static-typescript/

Maily this:

"Hello World" compiled from rust - 406 KB
"Hello World" compiled from deno - 118.3 MB

My question is why isn't the compiler you've built for device script also accessible as a stand-alone for typescript in general, I don't think people know what you've built here?! Is there a compiler available without the use of devicescript vscode plugin, boards and such? If so I wasn't able to see this in the documentation. `npx devicescript compile example.ts` and output the bundle.

Tried something like this `npx @devicescript/cli bundle --board pico_w example.ts` but it's bundle didn't run 😆 

Am I missing something as to why this isn't within scope for this project?

**Comment by mmoskal** [comment]
The runtime we have built is targeted at small embedded devices, with under 1MB of RAM. In particular, strings and arrays are limited to 64kB. This is the kind of thing you tend to hit pretty quickly...

If you want small JS interpreter, you could try quickjs.
